chromadb/api/fastapi.py

```diff
@@ -31,21 +31,28 @@ def __init__(self, system: System):
         super().__init__(system)
         url_prefix = "https" if system.settings.chroma_server_ssl_enabled else "http"
         system.settings.require("chroma_server_host")
-        system.settings.require("chroma_server_http_port")
-        self._api_url = f"{url_prefix}://{system.settings.chroma_server_host}:{system.settings.chroma_server_http_port}/api/v1"
+        if system.settings.chroma_server_http_port is not None:
+            self._api_url = f"{url_prefix}://{system.settings.chroma_server_host}:{system.settings.chroma_server_http_port}/api/v1"
+        else:  
+            self._api_url = f"{url_prefix}://{system.settings.chroma_server_host}/api/v1"
+        print(self._api_url)
         self._telemetry_client = self.require(Telemetry)
-
+        self._header=system.settings.chroma_server_header
+        self._session=requests.Session()
```

Comment:
fix formatting with spacing

do you know all the pros and cons of using `Session()`? Are there any cons or weird "gotchas"?

Reply:
`Session()` allow me to add the header in a cleaner way but as we reinitialise session on each requests i don't think there are any big cons